smac/initial_design/initial_design.py

```diff
@@ -118,118 +82,38 @@ def __init__(self,
             self.init_budget = len(configs)
         elif n_configs_x_params is not None:
             self.init_budget = int(max(1, min(n_configs_x_params * n_params,
-                                              (max_config_fracs * scenario.ta_run_limit))))
+                                              (max_config_fracs * ta_run_limit))))
         else:
             raise ValueError('Need to provide either argument `init_budget`, `configs` or '
                              '`n_configs_x_params`, but provided none of them.')
-        if self.init_budget > scenario.ta_run_limit:
+        if self.init_budget > ta_run_limit:
             raise ValueError(
                 'Initial budget %d cannot be higher than the run limit %d.'
-                % (init_budget, scenario.ta_run_limit)
+                % (init_budget, ta_run_limit)
             )
         self.logger.info("Running initial design for %d configurations" % self.init_budget)
 
     def select_configurations(self) -> typing.List[Configuration]:
 
         if self.configs is None:
-            configs = self._select_configurations()
-            # add random configurations if _select_configurations() returns less than expected budget
-            if self.fill_random_configs and len(configs) < self.init_budget:
-                random_configs = self.scenario.cs.sample_configuration(size=self.init_budget - len(configs))
-                if isinstance(random_configs, list):
-                    configs.extend(random_configs)
-                else:
-                    configs.append(random_configs)
-            return configs
-        else:
-            return self.configs
-
-    def _select_configurations(self) -> typing.List[Configuration]:
-        raise NotImplementedError
+            self.configs = self._select_configurations()
 
-    def run(self) -> Configuration:
-        """Run the initial design.
-
-        Returns
-        -------
-        incumbent: Configuration
-            Initial incumbent configuration
-        """
-        configs = self.select_configurations()
-        for config in configs:
+        for config in self.configs:
             if config.origin is None:
                 config.origin = 'Initial design'
 
         # add this incumbent right away to have an entry to time point 0
         self.traj_logger.add_entry(train_perf=2**31,
```

Comment:
Can this be moved to the start() function of the SMBO class? I think it would make sense to have it closer to where runs are conducted, and it just does not fit here.

Reply:
It would also remove the last argument not directly related to the initial design out of this class.

Reply:
`traj_logger` is not a part of SMBO since they are not directly related either. So, it might be better if it remained at `InitialDesign`?

Reply:
Hm, good point. However, it would reduce the amount of arguments that need to be passed to the initial design, and I think make the overall design a bit clearer as the initial design will have only a single responsibility.

Reply:
Given the EPILS facade I see the advantage of having the logger in the initial design.